test/integration/asset_manager_test.rb

```diff
@@ -10,33 +12,73 @@ class CreatingAFileAttachment < ActiveSupport::TestCase
       )
     end
 
-    test "sends the attachment to Asset Manager" do
-      Services.asset_manager.expects(:create_whitehall_asset).with(file_and_legacy_url_path_matching(/#{@filename}/))
+    describe "use_non_legacy_endpoints is false" do
+      test "sends the attachment to Asset Manager" do
+        Services.asset_manager.expects(:create_whitehall_asset).with(file_and_legacy_url_path_matching(/#{@filename}/))
 
-      Sidekiq::Testing.inline! do
-        @attachment.save!
+        Sidekiq::Testing.inline! do
+          @attachment.save!
+        end
```

Comment:
I wonder if this is giving us a false sense of security that the `AttachmentData` will already have been created, despite the code comment saying otherwise.